app/Http/Controllers/ImportController.php

```diff
@@ -34,4 +35,93 @@ public function store(Request $request) {
 
         return redirect()->route('imports.index');
     }
+
+    public function getPrepare(Import $import) {
+        $this->authorize('modify', $import);
+
+        $headers = [];
+
+        $file = fopen(storage_path('app/imports/' . $import->file), 'r');
```

Comment:
No. No. No. No. Fucking no.

You don't store user submitted files in the app directory. You use storage/files at the very least. In a place where your users won't be able to read this file on their own.